### ISSUE TITLE
Glossary page is corrupted

### ISSUE THREAD

**Issue by ehuss** [rust-lang/rustc-dev-guide#1041]
The [glossary page](https://rustc-dev-guide.rust-lang.org/appendix/glossary.html) is not rendering correctly.  Currently, it looks like this:

<img width="762" alt="image" src="https://user-images.githubusercontent.com/43198/106075579-bd40e200-60c2-11eb-91ba-3bdff26bef1e.png">

It looks like this is caused by mdbook-toc.  Looking at the implementation, it looks like it is using pulldown-cmark-to-cmark, which in my experience is not very reliable.

(EDIT: no aspersions meant for pulldown-cmark-to-cmark, it is a very hard problem to convert the events back to markdown.)

**Comment by igaray** [comment]
I think I found the problem. 

The `<span>` element in the tables is triggering pulldown-cmark-to-cmark to include a spurious newline, which causes the following table cell to be re-rendered in markdown as a new row. 

This change was introduced in pulldown-cmark-to-cmark [here](https://github.com/Byron/pulldown-cmark-to-cmark/pull/11)

Still thinking what we can do about it.

**Comment by camelid** [comment]
This doesn't solve the underlying problem, but we could change the layout of the Glossary to be like:

```markdown
#### arena/arena allocation

... info about arenas

#### TyCtxt

... info about TyCtxt
```

which should "solve" the issue. I think the current Glossary layout is hard to read and navigate as it is, this approach would mean:

- We can get of the `span` tags, which are kind of a hacky approach
- We can more easily link directly to glossary items
- The glossary would be 100% pure Markdown, no HTML
- It would be easier to add items to the glossary
- It would be easier to read the glossary

So, why not ;)

**Comment by igaray** [comment]
I like this approach, if there's consensus I'll report the root cause upstream and "fix" ours this way.

**Comment by ehuss** [comment]
That is how it is done with [Cargo](https://doc.rust-lang.org/cargo/appendix/glossary.html) and the [reference](https://doc.rust-lang.org/reference/glossary.html). It takes up a lot more space, but overall I think it works.

**Comment by VitalyAnkh** [comment]
Could I make a PR to solve this in @ehuss 's approach?

**Comment by igaray** [comment]
@VitalyAnkh absolutely! I had started on this but hadn't gotten to fixing all the references in the rest of the book. Go ahead!

**Comment by VitalyAnkh** [comment]
I find it's hard to add link without `<span>` element..

**Comment by LeSeulArtichaut** [comment]
I think `mdbook` creates an ID for each title element?

**Comment by VitalyAnkh** [comment]
@LeSeulArtichaut Thanks!  `<span>`s removed!

**Comment by rylev** [comment]
This has been fixed by #1146.
